WMIAdventure/backend/WMIAdventure_backend/proposed_content/views.py

```diff
@@ -1,9 +1,49 @@
-from rest_framework import generics
+from rest_framework import generics, status
+from rest_framework.response import Response
+from rest_framework.views import APIView
 
+from cards.models import CardInfo
+from cards.serializers import WholeCardSerializer
 from proposed_content.models import ProposedCardInfo
 from proposed_content.serializers import WholeProposedCardSerializer
 
 
+class AcceptProposedCardView(APIView):
+    """
+    post:
+
+    Accept proposed card with given id.
+    """
+
+    def post(self, request, pk):
+        try:
+            # Get proposed card that will be accepted
+            proposed_card = ProposedCardInfo.objects.get(pk=pk)
+        except ProposedCardInfo.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        # Serialize proposed cards data
+        proposed_card_serializer = WholeProposedCardSerializer(instance=proposed_card)
+
+        try:
+            # If card with proposed card's name exists - update existing card with proposed card's data
+            card_to_update = CardInfo.objects.get(name=proposed_card.name)
+            accepted_card_serializer = WholeCardSerializer(instance=card_to_update, data=proposed_card_serializer.data)
+            response_status = status.HTTP_200_OK
```

Comment:
Nie jestem pewien, czy ustawianie tutaj statusu na OK jest słuszne. Po bloku try/except mamy walidację danych w serializerze, wtedy może pójść coś nie tak. W normalnych warunkach takie coś nie przejdzie, bo przecież bierzemy karty z bazy obiektów, więc musi przejść walidację. ALE! W przyszłości możemy zaktualizować karty, np dodać im jakieś nowe pole, które będzie potrzebne.

Załóżmy, że ktoś stworzył kartę przed aktualizacją.

Aktualizujemy model i potrzebujemy tam mieć jakieś nowe pole.

Po aktualizacji mamy obiekt, który ma w bazie nulle w tym miejscu gdzie jest nowe pole. Serializer może np nie chcieć nulli - wtedy powinniśmy nie mieć możliwości zaakceptowania karty. Tutaj damy HTTP_OK a w środku wywali się jakiś internal server error.

Reply:
![image](https://user-images.githubusercontent.com/43492764/121801277-d3e17300-cc36-11eb-94dc-8a64f664452c.png)

Jeśli serializer nie będzie chciał nulli to wyrzuci validation error i widok automatycznie zróci 400 Bad Request

![image](https://user-images.githubusercontent.com/43492764/121801318-0f7c3d00-cc37-11eb-865a-0d9550574296.png)